src/gui_common/AddWindowReorderingSupportToSiblings.cs

```diff
@@ -0,0 +1,328 @@
+﻿using System.Collections.Generic;
+using Godot;
+using Godot.Collections;
+using static CustomWindow;
+
+/// <summary>
+///   Reorders windows by reordering their ancestors that are on the same level as this node.
+/// </summary>
+/// <remarks>
+///   <para>
+///     The windows are the ones who ask to establish the connection. If the window asks for a recursive
+///     connection then the connection gets passed to window reordering nodes in ancestors too.
+///   </para>
+/// </remarks>
+public class AddWindowReorderingSupportToSiblings : Control
+{
+    /// <summary>
+    ///   Paths to window reordering nodes in ancestors.
+    /// </summary>
+    [Export]
+    public Array<NodePath> WindowReorderingPaths = new();
+
+    /// <summary>
+    ///   Finds first window reordering node in ancestors to connect to.
```

Comment:
This should clarify that this controls the search depth as that's the point of this variable, right now this comment is still pretty unclear.